code/src/Components/MessageList.js

```diff
@@ -0,0 +1,59 @@
+import React, { useState } from 'react';
+
+import Message from './Message';
+
+const MessageList = ({ listOfMessages, setListOfMessages }) => {
+    const [likedPosts, setLikedPosts] = useState([]);
+    
+    const whenLiked = (id) => {
+        // Maps over the current messages and increments the like number
+        const updatedMessagesWithLikes = listOfMessages.map(message => {
+            if (message._id === id) {
+                message.hearts += 1
+            }
+            return message
+        })
+        // Changes the state with array with updated likes no
+        setListOfMessages(updatedMessagesWithLikes)
+
+        // Checks if postID is in array - if not - add to array
+        if (!likedPosts.includes(id)) {
+            setLikedPosts([...likedPosts, id])
+        }
+    }
+    
+    return (
+        <div className="message-list">
+            {
+                listOfMessages.map((message) => {
+                    return (
+                        <Message
+                            key={message._id}
+                            id={message._id}
```

Comment:
What you actually could do is pretty similar to what you did above with:

<Message {...message} />
and later on in the js file where you render everything to the DOM - you could just print out 

message,
createdAt,
hearts, 

as parameters (as we did for this weeks move-site project)